src/sortedDict.jl

```diff
@@ -79,336 +58,50 @@ end
-end
-
-
-
-function colon{K, D, Ord <: Ordering}(i1::SDToken{K,D,Ord}, i2::SDToken{K,D,Ord})
-    !(i1.container === i2.container) &&
-        throw(ArgumentError("SDToken range constructor requires tokens for the same container"))
-    IncludeLast(i1.container, i1.semitoken.address, i2.semitoken.address)
-end
-
-
-# need the next two to resolve ambiguity with keys(Associative)
-keys(m::SortedDict) = SDKeyIteration(m)
-values(m::SortedDict) = SDValIteration(m)
-keys{T <: SDIterableTypesBase}(ba::T) = SDKeyIteration(ba)
-values{T <: SDIterableTypesBase}(ba::T) = SDValIteration(ba)
-tokens{T <: SDIterableTypesBase}(ba::T) = SDTokenIteration(ba)
-tokens{T <: SDIterableTypesBase}(ki::SDKeyIteration{T}) = SDTokenKeyIteration(ki.base)
-tokens{T <: SDIterableTypesBase}(vi::SDValIteration{T}) = SDTokenValIteration(vi.base)
-
-start(m::SortedDict) = SDIterationState(m, nextloc0(m.bt,1), 2)
-
-function start(e::ExcludeLast) 
-    (!(e.first in e.m.bt.useddatacells) || e.first == 1 ||
-        !(e.pastlast in e.m.bt.useddatacells)) &&
-        throw(BoundsError())
-    if compareInd(e.m.bt, e.first, e.pastlast) < 0
-        return SDIterationState(e.m, e.first, e.pastlast) 
-    else
-        return SDIterationState(e.m, 2, 2)
-    end
-end
-
-function start(e::IncludeLast) 
-    (!(e.first in e.m.bt.useddatacells) || e.first == 1 ||
-        !(e.last in e.m.bt.useddatacells) || e.last == 2) && 
-        throw(BoundsError())
-    if compareInd(e.m.bt, e.first, e.last) <= 0
-        return SDIterationState(e.m, e.first, nextloc0(e.m.bt, e.last)) 
-    else
-        return SDIterationState(e.m, 2, 2)
-    end
-end
-
-start(e::SDCompoundIterable) = start(e.base)
-
-function isless{K, D, Ord <: Ordering}(s::SDToken{K,D,Ord}, t::SDToken{K,D,Ord})
-    !(s.container === t.container) &&  
-        throw(ArgumentError("SDToken isless requires tokens for the same container"))
-    return compareInd(s.container.bt, 
-                      s.semitoken.address, 
-                      t.semitoken.address) < 0
-end
-
-
-function isequal{K, D, Ord <: Ordering}(s::SDToken{K,D,Ord}, t::SDToken{K,D,Ord})
-    !(s.container === t.container) && 
-        throw(ArgumentError("SDToken isequal requires tokens for the same container"))
-    return s.semitoken.address == t.semitoken.address
-end
-
-
-function in{K,D,Ord <: Ordering}(pr::(@compat Tuple{Any,Any}), m::SortedDict{K,D,Ord})
+@inline function in{K,D,Ord <: Ordering}(pr::(@compat Tuple{Any,Any}), m::SortedDict{K,D,Ord})
     i, exactfound = findkey(m.bt,convert(K,pr[1]))
     return exactfound && isequal(m.bt.data[i].d,convert(D,pr[2]))
 end
 
-eltype{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}) = (K,D)
+@inline eltype{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}) = @compat Tuple{K,D}
+@inline orderobject(m::SortedDict) = m.bt.ord
+@inline keytype{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}) = K
+@inline datatype{K,D,Ord <: Ordering}(m::SortedDict{K,D,Ord}) = D
```

Comment:
Even if you make the changes I suggested, I did see that this might be useful when using typealiases, such as `SDMContainer`, since those aren't parameterized.  I'll note that these can be written as 

```julia
@inline keytype{K}(m::SortedDict{K}) = K
@inline datatype{K,D}(m::SortedDict{K,D}) = D
```

Minor: For consistency with the other dictionary types (here and in `Base`), the `D` type parameter is typically written as `V` for value.  But I don't care if that is changed here--I'd rather get this finished and in soon.